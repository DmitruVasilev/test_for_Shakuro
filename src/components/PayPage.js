import React from "react";
import { Link, Redirect } from "react-router-dom";
import PropTypes from "prop-types";
import { operators } from "../mock-data";
import PayCaption from "./PayCaption";
import PayForm from "./PayForm";

class PayPage extends React.Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        operatorId: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  };

  state = {
    redirect: false
  };

  handleRedirect = isRedirect => {
    this.setState({ redirect: isRedirect });
  };

  render() {
    const { redirect } = this.state;
    const operator = operators.filter(
      operator => operator.id === this.props.match.params.operatorId
    )[0];
    if (redirect) {
      return <Redirect to="/" />;
    }
    return (
      <React.Fragment>
        <Link to="/">Назад</Link>
        <PayCaption icon_url={operator.icon_url} title={operator.title} />
        <PayForm handleRedirect={this.handleRedirect} />
      </React.Fragment>
    );
  }
}

export default PayPage;
