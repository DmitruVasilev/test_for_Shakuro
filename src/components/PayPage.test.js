/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import PayPage from "./PayPage";

const matchMock = {
  match: {
    params: { operatorId: "1" }
  }
};

describe("<PayPage />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <PayPage {...matchMock} />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <PayPage {...matchMock} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
