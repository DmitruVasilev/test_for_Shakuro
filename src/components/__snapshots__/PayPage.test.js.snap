// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PayPage /> renders correctly 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
  >
    Назад
  </a>,
  <h1
    className="sc-bdVaJa lmOnTH"
  >
    <img
      alt="MTS"
      src="https://citycard.ru/i/landing-features/operator-mts.svg"
    />
    <span
      className="sc-bwzfXH cgXZhI"
    >
      MTS
    </span>
  </h1>,
  <form
    className="sc-htpNat bhpHJD"
    onSubmit={[Function]}
  >
    <label
      className="sc-EHOje kDcnGL"
      htmlFor="phone"
    >
      Введите номер телефона:
      <br />
      <input
        className="sc-ifAKCX duHYHE"
        id="phone"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        type="phone"
        value=""
      />
    </label>
    <span
      className="sc-bZQynM hhxum"
    >
      
    </span>
    <label
      className="sc-EHOje kDcnGL"
      htmlFor="number"
    >
      Введите сумму:
      <br />
      <input
        className="sc-bxivhb jMoerp"
        id="number"
        onChange={[Function]}
        type="number"
        value=""
      />
      <span
        className="sc-gzVnrw eaBHMQ"
      >
        ₽
      </span>
    </label>
    <span
      className="sc-bZQynM hhxum"
    >
      
    </span>
    <button
      className="sc-htoDjs kQZvSM"
      disabled=""
      type="submit"
    >
      Оплатить
    </button>
    
  </form>,
]
`;
